src/libraries/System.IO.FileSystem.Watcher/tests/Utility/FileSystemWatcherTest.cs

```diff
@@ -607,5 +602,25 @@ void AddEvent(WatcherChangeTypes eventType, string dir1, string dir2 = "")
                 }
             }
         }
+
+        internal class TestISynchronizeInvoke : ISynchronizeInvoke
+        {
+            public bool BeginInvoke_Called;
+            public Delegate ExpectedDelegate;
+
+            public IAsyncResult BeginInvoke(Delegate method, object[] args)
+            {
+                if (ExpectedDelegate != null)
+                    Assert.Equal(ExpectedDelegate, method);
+
+                BeginInvoke_Called = true;
+                method.DynamicInvoke(args[0], args[1]);
+                return null;
+            }
```

Comment:
> Are the `TestISynchronizeInvoke` delegates, such as the following, supposed to be invoked?
> 
> https://github.com/dotnet/runtime/blob/6474c725025a8b29b4265c4d9485018982ee6b16/src/libraries/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.cs#L121-L137
> 
> If so, should [`TestISynchronizeInvoke.BeginInvoke()`](https://github.com/dotnet/runtime/blob/6474c725025a8b29b4265c4d9485018982ee6b16/src/libraries/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.cs#L60-L65) be changed to something like the below?
> 
> ```cs
> public IAsyncResult BeginInvoke(Delegate method, object[] args)
> {
>     Assert.Equal(ExpectedDelegate, method);
>     BeginInvoke_Called = true;
>     method.DynamicInvoke(args[0], args[1]);
>     return null;
> }
> ```

I changed this to something like the above, so the following handler would get invoked:

https://github.com/dotnet/runtime/blob/9fa24a29c84ddb49492ca9ce7cfcc9bda82ff264/src/libraries/System.IO.FileSystem.Watcher/tests/Utility/FileSystemWatcherTest.cs#L399-L403